### PR TITLE
SRCH-3876: Remove therubyracer. Use cssbuilding-rails gem and less node package instead

### PR DESCRIPTION
**Pull request by stevenbarragan**
## Summary
- Replace the `therubyracer`, `less`, and `less-rails-bootstrap` gems with the `cssbuilding-rails` gem and the `less` package from node directly.
- Copy bootstrap `less` files from `less-rails-bootstrap` into search-gov vendor's folder
- Developers will need to run the `bin/dev` script to start the rails server and the process to watch and compile `less` files together locally.
- Alternative developers can run just the `yarn run watch` command separately from the rails server as well.
- `cssbuilding-rails` integrates with the rails assets pipeline and runs `yarn run build:css` automatically during assets pre-compilation in production and staging environments.

Changes in https://otter.staging.search.usa.gov/admin

### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Review by lorawoodford — CHANGES_REQUESTED** [review]
@stevenbarragan  This is looking really great.  I left a few minor comments inline.

But I noticed a blocker just now and wanted to report it asap before finishing this review entirely.  There seems to be an issue picking up some of the twitter bootstrap icons that I'm seeing both locally and in your otter stack.  This is most obvious if you go to a preview search page for any affiliate and look at the (default) blue submit button.  In prod it grabs a white magnifying glass icon, but locally and on otter we're not getting that icon.  Happy to dig more shortly:
Screenshot 2023-07-31 at 2 05 51 PM

Thanks!

File: Gemfile
Comment (stevenbarragan) [reply Gemfile:123]: Created https://cm-jira.usa.gov/browse/SRCH-4407

File: README.md
Comment (stevenbarragan) [reply README.md:206]: Running `assets:precompile` is needed only for production environments. I added a note at the end of this README but it might be better for it to be documented in the `search-infra` repo.

File: app/assets/stylesheets/sites_ie.css
Comment (stevenbarragan) [reply app/assets/stylesheets/sites_ie.css:1]: Yes! That was the whole point of this change. I just deleted it.

**Comment by stevenbarragan** [comment]
@lorawoodford I figure out how to [solve](https://github.com/GSA/search-gov/pull/1261/commits/237313ff030ea9d37563df996399bd1bae9d707a) the issue with the `asset-url`. It's working on the otter stack now.

**Comment by lorawoodford** [comment]
@stevenbarragan I've been going around on circles with this this morning locally because I'm now blocked from just about everything due to having zscaler sprung on me last night, but didn't want to leave this hanging out here.  Perhaps someone with a functioning computer like @ddzz or @krbhavith can give this an additional look.

That said, my *gut* is that while this definitely works I feel like we shouldn't technically _need_ this since it's been addressed upstream in sprockets-rails: https://github.com/rails/sprockets-rails/pull/476.  A small part of me feels we're kicking some maintenance pain down the road a bit here, but I'm inclined to ignore that for now because this set of changes alone is a huge leap forward.  Ideally I'd try out a few things here locally myself, but seeing as I can't even `yarn install` without it getting blocked by zscaler, that's probably not going to happen in a timely manner.  

Happy to add an approval once the merge conflict is addressed and once some other person with a working machine give it a quick check.  👍

File: bin/css
Comment (ddzz) [inline bin/css:1]: Do we need to run this is staging and production? Or will `assets:precompile` handle it?

**Comment by stevenbarragan** [comment]
@lorawoodford another `AssetUrlProcessor` is needed because the one from [lib/sprockets/rails/asset_url_processor.rb](https://github.com/rails/sprockets-rails/pull/476/files#diff-163243a09273299560b3deed2d4f898258c93d76ac91d4d40e70af8568be458f) only works with the `url` parameter instead of the `asset-url` like we need it to.

**Review by lorawoodford — APPROVED** [review]
@stevenbarragan Leaving an approval for this, but we should carve out time to discuss test plans next week at the end of daily scrum as this has (as has already been discussed) developer process impacts as well as (the potential for, though there shouldn't be any!) public-facing impact and, as such, requires fed Acceptance Testing by policy.

**Comment by lorawoodford** [comment]
@stevenbarragan be sure to update the JIRA number in the PR title here to reflect the current story/not the investigation story... that's how the PR gets automatically linked in JIRA.

**Pull request merged**
